parflow_data_management/consumers.py

```diff
@@ -0,0 +1,31 @@
+import json
+
+from channels.generic.websocket import JsonWebsocketConsumer
+from asgiref.sync import async_to_sync
+
+class GeneralConsumer(JsonWebsocketConsumer):
+    def connect(self):
+        self.accept()
+        group = compute_group_for_user(self.scope["user"].id)
+        async_to_sync(self.channel_layer.group_add)(group, self.channel_name)
+
+    def disconnect(self, close_code):
+        group = compute_group_for_user(self.scope["user"].id)
+        async_to_sync(self.channel_layer.group_discard)(group, self.channel_name)
+
+    def public_key_with_passphrase(self, message):
+        self.send_json({
+            "type": "public.key.with.passphrase",
+            "data": message["data"],
+        })
```

Comment:
When we send a message to a consumer, channels matches the `type` field of the message with the name of a method in the consumer to figure out which method will handle the message. For example, to execute a remote command, we call `channel_layer.group_send(group, {"type": "command.output", "data": data})`. Channels then looks for a method in the general consumer called `command_output` (replaces dots with underscores). Because of this, we would need to move the entire consumer or split it into 2 consumers.

Reply:
It seems we could decouple this by registering these callback with `GeneralConsumer` and just use the type to look up the appropriate function, thus removing the application specific code from the consumer?

Reply:
How would we do this exactly? With subtypes? The methods need to be in a consumer class.

Reply:
Name changed in 6eae592e55f2225926b30404485537c18f43c0b0.